### ISSUE TITLE
Latest 4.0.0 Eclipse plugin is not functional

### ISSUE THREAD

**Issue by TheRealAyCe** [spotbugs/spotbugs#1067]
- The views no longer work
- Find bugs option disappeared from the context menu of a project
- ~~Marketplace reports it has version 3.1.5.r201806132012-cbbf0a5 installed (:?)~~ reinstalled from the site
- (Still) added in software sites and enabled: https://spotbugs.github.io/eclipse/
- In installation details:   SpotBugs	4.0.0.202001200351-6732852 com.github.spotbugs.plugin.eclipse.feature.group	SpotBugs Project
- No SpotBugs warnings/errors in the Problems view anymore (don't remember if there were any)

!ENTRY org.eclipse.jface 4 0 2020-01-21 11:11:42.753
!MESSAGE Unable to resolve plug-in "com.github.spotbugs.plugin.eclipse".
!STACK 0
java.io.IOException: Unable to resolve plug-in "com.github.spotbugs.plugin.eclipse".
	at org.eclipse.core.internal.runtime.PlatformURLPluginConnection.parse(PlatformURLPluginConnection.java:64)
	at org.eclipse.core.internal.runtime.PlatformURLPluginConnection.resolve(PlatformURLPluginConnection.java:73)
	at org.eclipse.core.internal.boot.PlatformURLHandler.openConnection(PlatformURLHandler.java:69)
	at org.eclipse.osgi.internal.url.URLStreamHandlerProxy.openConnection(URLStreamHandlerProxy.java:123)
	at java.base/java.net.URL.openConnection(URL.java:1051)
	at org.eclipse.core.internal.runtime.PlatformURLConverter.toFileURL(PlatformURLConverter.java:34)
	at org.eclipse.core.runtime.FileLocator.toFileURL(FileLocator.java:261)
	at org.eclipse.jface.resource.URLImageDescriptor.getFilePath(URLImageDescriptor.java:232)
	at org.eclipse.jface.resource.URLImageDescriptor.access$2(URLImageDescriptor.java:220)
	at org.eclipse.jface.resource.URLImageDescriptor$URLImageFileNameProvider.getImagePath(URLImageDescriptor.java:57)
	at org.eclipse.swt.internal.DPIUtil.validateAndGetImagePathAtZoom(DPIUtil.java:419)
	at org.eclipse.swt.graphics.Image.<init>(Image.java:687)
	at org.eclipse.jface.resource.URLImageDescriptor.createImage(URLImageDescriptor.java:263)
	at org.eclipse.jface.resource.ImageDescriptor.createResource(ImageDescriptor.java:181)
	at org.eclipse.jface.resource.DeviceResourceManager.allocate(DeviceResourceManager.java:55)
	at org.eclipse.jface.resource.AbstractResourceManager.create(AbstractResourceManager.java:88)
	at org.eclipse.jface.resource.ResourceManager.createImageWithDefault(ResourceManager.java:195)
	at org.eclipse.jface.resource.ImageRegistry.get(ImageRegistry.java:206)
	at org.eclipse.e4.ui.workbench.renderers.swt.SWTPartRenderer.getImageFromURI(SWTPartRenderer.java:220)
	at org.eclipse.e4.ui.workbench.renderers.swt.SWTPartRenderer.getImage(SWTPartRenderer.java:230)
	at org.eclipse.e4.ui.workbench.renderers.swt.StackRenderer.createTab(StackRenderer.java:880)
	at org.eclipse.e4.ui.workbench.renderers.swt.LazyStackRenderer.processContents(LazyStackRenderer.java:145)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeCreateGui(PartRenderingEngine.java:676)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeCreateGui(PartRenderingEngine.java:768)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.access$0(PartRenderingEngine.java:739)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$2.run(PartRenderingEngine.java:733)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.createGui(PartRenderingEngine.java:717)
	at org.eclipse.e4.ui.workbench.renderers.swt.SWTPartRenderer.processContents(SWTPartRenderer.java:72)
	at org.eclipse.e4.ui.workbench.renderers.swt.SashRenderer.processContents(SashRenderer.java:140)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeCreateGui(PartRenderingEngine.java:676)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeCreateGui(PartRenderingEngine.java:768)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.access$0(PartRenderingEngine.java:739)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$2.run(PartRenderingEngine.java:733)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.createGui(PartRenderingEngine.java:717)
	at org.eclipse.e4.ui.workbench.renderers.swt.SWTPartRenderer.processContents(SWTPartRenderer.java:72)
	at org.eclipse.e4.ui.workbench.renderers.swt.SashRenderer.processContents(SashRenderer.java:140)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeCreateGui(PartRenderingEngine.java:676)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeCreateGui(PartRenderingEngine.java:768)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.access$0(PartRenderingEngine.java:739)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$2.run(PartRenderingEngine.java:733)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.createGui(PartRenderingEngine.java:717)
	at org.eclipse.e4.ui.workbench.renderers.swt.SWTPartRenderer.processContents(SWTPartRenderer.java:72)
	at org.eclipse.e4.ui.workbench.renderers.swt.PerspectiveRenderer.processContents(PerspectiveRenderer.java:51)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeCreateGui(PartRenderingEngine.java:676)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeCreateGui(PartRenderingEngine.java:768)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.access$0(PartRenderingEngine.java:739)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$2.run(PartRenderingEngine.java:733)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.createGui(PartRenderingEngine.java:717)
	at org.eclipse.e4.ui.workbench.renderers.swt.PerspectiveStackRenderer.showTab(PerspectiveStackRenderer.java:82)
	at org.eclipse.e4.ui.workbench.renderers.swt.LazyStackRenderer.postProcess(LazyStackRenderer.java:105)
	at org.eclipse.e4.ui.workbench.renderers.swt.PerspectiveStackRenderer.postProcess(PerspectiveStackRenderer.java:64)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeCreateGui(PartRenderingEngine.java:680)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeCreateGui(PartRenderingEngine.java:768)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.access$0(PartRenderingEngine.java:739)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$2.run(PartRenderingEngine.java:733)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.createGui(PartRenderingEngine.java:717)
	at org.eclipse.e4.ui.workbench.renderers.swt.SWTPartRenderer.processContents(SWTPartRenderer.java:72)
	at org.eclipse.e4.ui.workbench.renderers.swt.SashRenderer.processContents(SashRenderer.java:140)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeCreateGui(PartRenderingEngine.java:676)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeCreateGui(PartRenderingEngine.java:768)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.access$0(PartRenderingEngine.java:739)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$2.run(PartRenderingEngine.java:733)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.createGui(PartRenderingEngine.java:717)
	at org.eclipse.e4.ui.workbench.renderers.swt.SWTPartRenderer.processContents(SWTPartRenderer.java:72)
	at org.eclipse.e4.ui.workbench.renderers.swt.WBWRenderer.processContents(WBWRenderer.java:665)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeCreateGui(PartRenderingEngine.java:676)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeCreateGui(PartRenderingEngine.java:768)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.access$0(PartRenderingEngine.java:739)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$2.run(PartRenderingEngine.java:733)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.createGui(PartRenderingEngine.java:717)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1086)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1049)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:155)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:660)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:338)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:559)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:154)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:150)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:203)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:137)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:107)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:401)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:657)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:594)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1465)

**Comment by ubivent-seuffert** [comment]
maybe related to  #1066 ?

**Comment by iloveeclipse** [comment]
Right, Spotbugs Eclipse plugin in the current version is not functional.
See https://spotbugs.github.io/eclipse/plugins/com.github.spotbugs.plugin.eclipse_4.0.0.202001200351-6732852.jar
I don't see /lib directory shipped with the bundle, where all the required libraries should be.

**Comment by iloveeclipse** [comment]
I've just tried to run ```./gradlew eclipse```

and I see this:

```
> Task :spotbugs:eclipseClasspath FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':spotbugs:eclipseClasspath'.
> Resolving dependency configuration 'testImplementation' is not allowed as it is defined as 'canBeResolved=false'.
  Instead, a resolvable ('canBeResolved=true') dependency configuration that extends 'testImplementation' should be resolved.
```

**Comment by iloveeclipse** [comment]
This is a regression from #1061. If I build on commit a28ed511df74aebe1d5c5ab43bda5f3b8b763c1c  everything works.

**Comment by KengoTODA** [comment]
This is caused by `configurations.testImplementation` that is [not resolvable](https://docs.gradle.org/current/userguide/java_library_plugin.html) in Gradle v6. Instead, we need to use `configurations.testRuntimeClasspath`.

**Comment by KengoTODA** [comment]
@iloveeclipse could you test `a371301125ef2fcf0328176d5123f9c18547c430` to confirm the fix? I have no eclipse in my local.

**Comment by iloveeclipse** [comment]
Looks better, but now I see this error in Eclipse: ```Build path contains duplicate entry: '.libs/gui' for project 'spotbugs-spotbugs'```
This is coming from generated `spotbugs/spotbugs/.classpath` file that contains two references:
```
	<classpathentry kind="lib" path=".libs/gui" exported="true">
		<attributes>
			<attribute name="gradle_used_by_scope" value="gui,guiTest,test"/>
			<attribute name="test" value="true"/>
		</attributes>
	</classpathentry>
	<classpathentry kind="lib" path=".libs/gui" exported="true">
		<attributes>
			<attribute name="gradle_used_by_scope" value="gui,guiTest,test"/>
			<attribute name="test" value="true"/>
		</attributes>
	</classpathentry>
```

If I manually remove one from the entries above, project build complains about `Project 'spotbugs-spotbugs' is missing required library: '.libs/gui'`.

After removing both entries, spotbugs project is compilable in Eclipse.

Now to the eclipsePlugin project: it needs compilable spotbugs project first, but after fixing errors as described above I still see errors because the `spotbugs/eclipsePlugin/META-INF/MANIFEST.MF` file seem to be wrongly generated: it looks like the `Bundle-ClassPath` part is broken:

```
Bundle-Version: 4.0.0.202001221120-a371301
Bundle-Name: SpotBugs
Bundle-ClassPath: .,spotbugs-plugin.jar,
Bundle-Activator: de.tobject.findbugs.FindbugsPlugin
```

but should be
```
Bundle-Version: 4.0.0.202001221129-a28ed51
Bundle-Name: SpotBugs
Bundle-ClassPath: .,spotbugs-plugin.jar,lib/spotbugs.jar,lib/spotbugs-
 annotations.jar,lib/jsr305-3.0.2.jar,lib/Saxon-HE-9.9.1-2.jar,lib/icu
 4j-63.1.jar,lib/slf4j-api-1.8.0-beta4.jar,lib/commons-lang-2.6.jar,li
 b/jaxen-1.1.6.jar,lib/dom4j-2.1.1.jar,lib/jcip-annotations-1.0.jar,li
 b/bcel-6.4.1.jar,lib/asm-7.3.1.jar,lib/asm-tree-7.3.1.jar,lib/asm-ana
 lysis-7.3.1.jar,lib/asm-util-7.3.1.jar,lib/asm-commons-7.3.1.jar
Bundle-Activator: de.tobject.findbugs.FindbugsPlugin
```

**Comment by iloveeclipse** [comment]
I have unfortunately no time now to debug this, but eclipsePlugin/lib should contain all the libraries we require, and I guess the build code that copied dependencies to this place isn't working anymore.

**Comment by EdgeSync** [comment]
Also facing this issue, 

Windows 10 Enterprise, Tried Different versions of Eclipse, 32bit and 64bit + different JDK's, all before I found the issue here.

Spotbugs installs fine, and restart of eclipse is done, it lists as installed in the marketplace but doesn't show up in eclipse.

Confirmed on another Windows 10 PC with an older version of Spotbugs, update from  market place breaks it, and same issue happens - shows as installed in the marketplace, not available in eclipse.

**Comment by reynolb** [comment]
Thanks @EdgeSync, that is a good workaround. I can confirm that manually installing the previous release linked above works on macOS Mohave, with Eclipse 2019-09 and JDK 12.

**Comment by KengoTODA** [comment]
Fixed `spotbugs/eclipsePlugin/META-INF/MANIFEST.MF` at `8f0b8dd6b830ddbe6ea6925fb68e992899126c9a`, thank you.

But I cannot reproduce problem around `spotbugs/spotbugs/.classpath` 🤔 
Sorry but could you try this version again?

**Comment by iloveeclipse** [comment]
The lib is now populated, but it contains almost all *Eclipse* bundles (from dependencies) too :-(
It should only include things spotbugs itself needs plus spotbugs core.

The spotbugs project itself seem to be OK now, but Eclipse plugin complains it can't see spotbugs own packages...

I've quickly hacked a workaround, forgive me if this is not the gradle way, but I find ever changing gradle API extremely confusing compared to old good ant times:

```
diff --git a/eclipsePlugin/build.gradle b/eclipsePlugin/build.gradle
index d99939f..6d9d975 100644
--- a/eclipsePlugin/build.gradle
+++ b/eclipsePlugin/build.gradle
@@ -93,11 +93,17 @@
 }
 
 task copyLibsForEclipse(type: Copy) {
-    dependsOn configurations.compileClasspath
-    from configurations.compileClasspath.files {
+    dependsOn project(':spotbugs').configurations.compileClasspath // not sure this is needed
+    from ('../spotbugs/.libs') {
       include "*.jar"
       exclude "*xml-apis*.jar"
     }
+    from ('../spotbugs/build/libs') {
+      include "*.jar"
+    }
+    from ('../spotbugs-annotations/build/libs') {
+      include "*.jar"
+    }
     into "lib"
 }
 
```

**Comment by KengoTODA** [comment]
It is not Gradle's problem, I believe. IMHO our building procedure is not so common and we applied too many hacks to the build script.

Maybe I can have a try to standardize it, but less motivation I have for now.

**Comment by iloveeclipse** [comment]
> It is not Gradle's problem, I believe. IMHO our building procedure is not so common

A build tool that only supports common cases is worthless :-)

We use "ancient" make & ant at work where we have *extremely* uncommon, 20 years old application with ~15.000.000 LOC written in C, C++, Java, Eclipse, ruby, python etc - without *ever* fixing build files because of the tool update. With gradle I see even now after you've updated to 6 it starts to complain about something that will not work in 7. Almost every gradle update causes some pain somewhere.

Anyway, can we just use your commits + my workaround and fix the 4.0.0 RC?

**Comment by iloveeclipse** [comment]
@KengoTODA : the Eclipse update site on https://spotbugs.github.io/eclipse-latest/ still provides broken plugin, even built on  4e10bc7. It only contains spotbugs-annotations.jar jar in the lib directory.

**Comment by iloveeclipse** [comment]
Looks like "updateManifest" task (that should copy libraries & update manifest) runs before spotbugs jars are created...

```
 ./gradlew assemble --dry-run
:compileJava SKIPPED
:processResources SKIPPED
:classes SKIPPED
:jar SKIPPED
:assemble SKIPPED
:eclipsePlugin:featureCandidateJar SKIPPED
:spotbugs-annotations:compileJava SKIPPED
:spotbugs:compileJava SKIPPED
:eclipsePlugin:compileJava SKIPPED
:eclipsePlugin:processResources SKIPPED
:eclipsePlugin:classes SKIPPED
:spotbugs-annotations:processResources SKIPPED
:spotbugs-annotations:classes SKIPPED
:spotbugs-annotations:jar SKIPPED
:eclipsePlugin:copyLibsForEclipse SKIPPED
:spotbugs:copyLibsForEclipse SKIPPED
:spotbugs:updateManifest SKIPPED
:eclipsePlugin:updateManifest SKIPPED
:eclipsePlugin:jar SKIPPED
:eclipsePlugin:pluginJar SKIPPED
:eclipsePlugin:pluginCandidateJar SKIPPED
:eclipsePlugin:siteCandidateHtml SKIPPED
:eclipsePlugin:siteCandidateXml SKIPPED
:eclipsePlugin:generateCandidateP2Metadata SKIPPED
:eclipsePlugin:featureJar SKIPPED
:eclipsePlugin:siteHtml SKIPPED
:eclipsePlugin:siteXml SKIPPED
:eclipsePlugin:generateP2Metadata SKIPPED
:eclipsePlugin:featureDailyJar SKIPPED
:eclipsePlugin:pluginDailyJar SKIPPED
:eclipsePlugin:siteDailyHtml SKIPPED
:eclipsePlugin:siteDailyXml SKIPPED
:eclipsePlugin:generateP2MetadataDaily SKIPPED
:eclipsePlugin:featureStableLatestJar SKIPPED
:eclipsePlugin:pluginStableLatestJar SKIPPED
:eclipsePlugin:siteStableLatestHtml SKIPPED
:eclipsePlugin:siteStableLatestXml SKIPPED
:eclipsePlugin:generateP2MetadataStableLatest SKIPPED
:eclipsePlugin:eclipseSite SKIPPED
:eclipsePlugin:eclipseSiteZip SKIPPED
:eclipsePlugin:assemble SKIPPED
:eclipsePlugin-junit:compileJava SKIPPED
:eclipsePlugin-junit:processResources SKIPPED
:eclipsePlugin-junit:classes SKIPPED
:eclipsePlugin-junit:jar SKIPPED
:eclipsePlugin-junit:assemble SKIPPED
:test-harness-core:compileJava SKIPPED
:test-harness:compileJava SKIPPED
:eclipsePlugin-test:compileJava SKIPPED
:eclipsePlugin-test:processResources SKIPPED
:eclipsePlugin-test:classes SKIPPED
:eclipsePlugin-test:jar SKIPPED
:eclipsePlugin-test:sourcesJar SKIPPED
:eclipsePlugin-test:assemble SKIPPED
:spotbugs:processResources SKIPPED
:spotbugs:classes SKIPPED
:spotbugs:compileGuiJava SKIPPED
:spotbugs:processGuiResources SKIPPED
:spotbugs:guiClasses SKIPPED
:spotbugs:jar SKIPPED
:spotbugs:scripts SKIPPED
:spotbugs-ant:compileJava SKIPPED
:spotbugs-ant:processResources SKIPPED
:spotbugs-ant:classes SKIPPED
:spotbugs-ant:jar SKIPPED
:spotbugs:distTar SKIPPED
:spotbugs:distZip SKIPPED
:spotbugs:assembleDist SKIPPED
:spotbugs:distSrcZip SKIPPED
:spotbugs:assemble SKIPPED
:spotbugs-annotations:assemble SKIPPED
:spotbugs-ant:assemble SKIPPED
... rest omitted ...
```

Same for before gradle 6 update:
```
 ./gradlew assemble --dry-run
:compileJava SKIPPED
:processResources SKIPPED
:classes SKIPPED
:jar SKIPPED
:assemble SKIPPED
:eclipsePlugin:featureCandidateJar SKIPPED
:spotbugs-annotations:compileJava SKIPPED
:spotbugs-annotations:processResources SKIPPED
:spotbugs-annotations:classes SKIPPED
:spotbugs-annotations:jar SKIPPED
:spotbugs:compileJava SKIPPED
:spotbugs:processResources SKIPPED
:spotbugs:classes SKIPPED
:spotbugs:compileGuiJava SKIPPED
:spotbugs:processGuiResources SKIPPED
:spotbugs:guiClasses SKIPPED
:spotbugs:jar SKIPPED
:eclipsePlugin:compileJava SKIPPED
:eclipsePlugin:processResources SKIPPED
:eclipsePlugin:classes SKIPPED
:eclipsePlugin:copyLibsForEclipse SKIPPED
:spotbugs:copyLibsForEclipse SKIPPED
:spotbugs:updateManifest SKIPPED
:eclipsePlugin:updateManifest SKIPPED
:eclipsePlugin:jar SKIPPED
:eclipsePlugin:pluginJar SKIPPED
:eclipsePlugin:pluginCandidateJar SKIPPED
:eclipsePlugin:siteCandidateHtml SKIPPED
:eclipsePlugin:siteCandidateXml SKIPPED
:eclipsePlugin:generateCandidateP2Metadata SKIPPED
:eclipsePlugin:featureJar SKIPPED
:eclipsePlugin:siteHtml SKIPPED
:eclipsePlugin:siteXml SKIPPED
:eclipsePlugin:generateP2Metadata SKIPPED
:eclipsePlugin:featureDailyJar SKIPPED
:eclipsePlugin:pluginDailyJar SKIPPED
:eclipsePlugin:siteDailyHtml SKIPPED
:eclipsePlugin:siteDailyXml SKIPPED
:eclipsePlugin:generateP2MetadataDaily SKIPPED
:eclipsePlugin:featureStableLatestJar SKIPPED
:eclipsePlugin:pluginStableLatestJar SKIPPED
:eclipsePlugin:siteStableLatestHtml SKIPPED
:eclipsePlugin:siteStableLatestXml SKIPPED
:eclipsePlugin:generateP2MetadataStableLatest SKIPPED
:eclipsePlugin:eclipseSite SKIPPED
:eclipsePlugin:eclipseSiteZip SKIPPED
:eclipsePlugin:assemble SKIPPED
... rest omitted ...
```

So before gradle 6 change, spotbugs:copyLibsForEclipse was running before we run eclipsePlugin:updateManifest.

**Comment by iloveeclipse** [comment]
OK, the last build result on https://spotbugs.github.io/eclipse-latest/ works fine.
All on ticket: feel free to update from this update site & report regressions against 4.0.0 RC1.

@KengoTODA : I assume the  https://spotbugs.github.io/eclipse-candidate/ and https://spotbugs.github.io/eclipse/ are updated on next RC build, right?

Should we trigger RC2 now, or is something else need to be fixed & merged first?

**Comment by TheRealAyCe** [comment]
Just installed it and tested it. Seems to work fine now. Already fixed some bugs I introduced while the plugin was offline. :P

**Comment by davidkarlsen** [comment]
Is this the last issue for 4.0 to be ready to be 🚢 ?
It has been holding back java13 support in Sonar for quite some time now: https://github.com/spotbugs/sonar-findbugs/issues/293

**Comment by iloveeclipse** [comment]
This one is actually fixed by #1071.
